api/core/v1beta2/cluster_types.go

```diff
@@ -976,6 +975,24 @@ type MachineDeploymentTopologyHealthCheckChecks struct {
 // MachineDeploymentTopologyHealthCheckRemediation configures if and how remediations are triggered if a MachineDeployment Machine is unhealthy.
 // +kubebuilder:validation:MinProperties=1
 type MachineDeploymentTopologyHealthCheckRemediation struct {
+	// maxInFlight determines how many in flight remediations should happen at the same time.
+	//
+	// Remediation only happens on the MachineSet with the most current revision, while
+	// older MachineSets (usually present during rollout operations) aren't allowed to remediate.
+	//
+	// Note: In general (independent of remediations), unhealthy machines are always
+	// prioritized during scale down operations over healthy ones.
+	//
+	// MaxInFlight can be set to a fixed number or a percentage.
+	// Example: when this is set to 20%, the MachineSet controller deletes at most 20% of
+	// the desired replicas.
+	//
+	// If not set, remediation is limited to all machines (bounded by replicas)
+	// under the active MachineSet's management.
+	//
+	// +optional
+	MaxInFlight *intstr.IntOrString `json:"maxInFlight,omitempty"`
```

Comment:
Have you thought about setting a MaxLength here? Controller-gen might complain but if you give it the intorstring marker as well it will allow you to set maxlength, which I guess would be 4? 

Do we have validation that when this is a string, that it is a percentage?